source/frontend/src/components/common/mapFSM/MapStateMachineContext.tsx

```diff
@@ -101,7 +101,7 @@ export const MapStateMachineProvider: React.FC<React.PropsWithChildren<unknown>>
         } else if (event.type === 'MAP_MARKER_CLICK') {
           latLng = event.featureSelected.latlng;
         }
-        const result = locationLoader.loadLocationDetails(latLng);
+        const result = locationLoader.loadLocationDetails(latLng, event.featureSelected);
```

Comment:
the loadLocationData presumes we are trying to load based on a click of a marker (lat-lng). If the problem is that the pims property is using the pid-pin to find it after searching for the parcel, then instead it should query the pims layer by the lat-lng instead of attempting to use the pims id

Reply:
I'll try that, was concerned about performance since the lat/lng is not indexed. Will do some testing in one of our more marker heavy envs to check performance.

Reply:
well, it would be doing that search on the other layers too, so assuming those are equal performant it should be fine. Also, as it is now, the pims call waits on the external layer to resolve the pid. If they are doing the call independently they could be parallelized.